core/FrontController.php

```diff
@@ -460,6 +460,10 @@ protected function prepareDispatch($module, $action, $parameters)
 
         if (is_null($action)) {
             $action = Common::getRequestVar('action', false);
+            if ($action !== false) {
+                // If a value was provided, check it has the correct type.
+                $action = Common::getRequestVar('action', null, 'string');
```

Comment:
This will throw an exception with the message `The parameter 'action' doesn't have a correct type, and a default value wasn't provided.`

Directly using `Piwik::getAction()` would in theory also work, but it would change the behavior slightly. Instead of throwing an error when an incorrect type was provided, it would use an empty string as default instead.